domain/storageprovisioning/filesystem.go

```diff
@@ -0,0 +1,29 @@
+// Copyright 2025 Canonical Ltd.
```

Comment:
Merge the two small files(filesystem.go and volume.go) into types.go?

Reply:
This is what has been asked of me in other PRs to split things out and not group.